### ISSUE TITLE
Дать возможность настройки сессии

### ISSUE THREAD

**Issue by tonal** [leshchenko1979/fast_bitrix24#140]
Мы используем BasicAuth при работе с коробочной версией.
Вполне возможно, что кому-то потребуется указать ещё какие-то дополнительные заголовки.

Хотелось бы иметь эту возможность.

П. С. Да, Сейчас довольно популярно использовать библиотеку [HTTPX](https://www.python-httpx.org/quickstart/) вместо [aiohttp](https://docs.aiohttp.org/en/stable/).
По API они существенно совместимы. Возможно имеет смысл дать возможность пользователю создавать свой объект сессий и его настраивать.

**Comment by leshchenko1979** [comment]
1. А можно какую-нибудь ссылочку про авторизацию в коробочной версии? А то все, что я могу найти, говорит, что там нужно делать OAUTH 2.0, а не BasicAuth.

2. А потребность именно в настройках сессии или в том, чтобы добавить один заголовок к запросу?

**Comment by tonal** [comment]
1. Это было сделано очень давно через изучение исходников и хак - тогда REST API в коробочной версии только появился - были сами файлы, но авторизация не работала.
Так что документации нет.

2. У [aiohttp.ClientSession](https://docs.aiohttp.org/en/stable/client_reference.html) есть параметр **auth** который формирует и подставляет нужный заголовок.
Так что пока нужна только возможность задавать значение **auth**.

**Comment by tonal** [comment]
Да, опять же экспериментальным путём было выяснено, что если запросы к битриксу (коробка не исключение) слать с базовой авторизацией реального пользователя, то они отрабатывают именно от этого пользователя.

**Comment by leshchenko1979** [comment]
А для чего делать, чтобы запросы отрабатывали от имени конкретного пользователя?

**Comment by tonal** [comment]
Они всё равно отрабатываются от какого-то пользователя.
И этот пользователь потом есть в истории изменений.
Кроме того, авторизация может так же отрабатываться до битрикса - в nginx-е например.

**Comment by leshchenko1979** [comment]
Я просто пытаюсь понять, насколько большому количеству пользователей это может понадобиться.

Пока что об этом попросили только вы, и это базируется на незадокументированных функциях.

Звучит как что-то, что должно делаться в форме, а не в основном репозитории. :)

**Comment by leshchenko1979** [comment]
В релизе 1.5.0 добавлена возможность использования пользовательского объекта клиентской сессии. Полагаю, этот issue можно закрывать.
